tests/pos/publicInBinary.scala

```diff
@@ -0,0 +1,140 @@
+//> using options -Werror -WunstableInlineAccessors
```

Comment:
Could this be a run test?

Reply:
Yes. When converting it to a run test I found a bug and fixed it.

We were accepting private constructor fields with no `val` because they had a privateWithin, But in this case we should only allow it is a `val` because we need the param accessor.
```scala
class Bar (
  @publicInBinary private[Bar] x: Int, // error (crash before fix)
  @publicInBinary private[Bar] val y: Int // ok
)
```
This is done in `Checking`.